editor/src/components/common/server-hooks.ts

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react'
+import { isLocal } from '../editor/persistence'
+import { reduxDevtoolsLogMessage } from '../../core/shared/redux-devtools'
+import { checkProjectOwnership, fetchProjectMetadata } from '../../common/server'
+
+type ProjectMetadata = {
+  title: string | null
+  ownerName: string | null
+  ownerPicture: string | null
+}
+
+export function useGetProjectMetadata(projectId: string | null): ProjectMetadata {
+  const previousProjectIdRef = React.useRef<string | null>(null)
+  const [userData, setUserData] = React.useState<ProjectMetadata>({
```

Comment:
I feel a little like this should be some kind of "unknown" value, as it's not that they're null, the first time we have no idea what the values are/could be.